src/main/java/org/apache/commons/lang3/RandomUtils.java

```diff
@@ -99,7 +102,7 @@ public static boolean nextBoolean() {
      *             {@code startInclusive} is negative
      * @return the random integer
      */
-    public static int nextInt(final int startInclusive, final int endExclusive) {
+    public static int nextInt(final @LessThan("#2") int startInclusive, final int endExclusive) {
```

Comment:
`startIndex` and the result should be `@NonNegative`. Similarly for the method below.